### PR TITLE
Use DYNAMIC on batch size

### PR DESCRIPTION
**Pull request by titaiwangms**
With the current regression on 20-ish LLMs, it's found that torch.export.Dim("batch", min=1, max=1024) sets a constraints to the graph when the tracing inputs have at least 2 on batch size. While with torch.export.Dim.DYNAMIC, such constraints are not set. From ONNX perspective, we don't have constraint range, so I think it makes sense that we only use DYNAMIC in such cases.

### CODE REVIEW

**Comment by titaiwangms** [comment]
[Check URLs / build (pull_request)](https://github.com/sdpython/onnx-diagnostic/actions/runs/17445952568/job/49540405554?pr=213) seems buggy to me.

**Review by justinchuby — APPROVED** [review]
